src/app/util/BUILD.gn

```diff
@@ -75,3 +75,11 @@ source_set("callbacks") {
     "${chip_root}/src/app:paths",
   ]
 }
+
+source_set("binding-table-headers") {
+  sources = [ "binding-table.h" ]
```

Comment:
we should not split cpp from header. Even this is what happens in practice, this allows for circular/odd dependencies. We should add source sets only when we are able to fix things.

Reply:
I agree in general, but I think this is a somewhat special case where we have an API that consists of some global functions that these cluster implementations can compile against, but the implementation of those functions relies on the application-level zap codegen. Without separating this we can't model this kind of link-time dependency (rather than a standard compile-time dependency).

I think being able to compile cluster implementations along with the core, and only requiring the generated code to be present at link time is a step forward from having to compile everything as part of the application's data model source set. Or is your new data model API work at a point where we can remove the dependency on zap-generated code entirely?

(Same story with the ember-api.h you've commented on below really. I would probably actually prefer to have a single header bundling all of these link-time dependencies on zap-generated code (for as long as we have to have them), and just treat that as a documented special case; as you say it's not a good pattern in general.)